src/library/trackmodel.h

```diff
@@ -1,54 +1,56 @@
 
-/** Pure virtual (abstract) class that provides an interface for data models which
-    display track lists. */
+/// Pure virtual (abstract) class that provides an interface for data models
+/// which display track lists.
 class TrackModel {
   public:
     static const int kHeaderWidthRole = Qt::UserRole + 0;
     static const int kHeaderNameRole = Qt::UserRole + 1;
 
     TrackModel(QSqlDatabase db,
-               const char* settingsNamespace)
+            const char* settingsNamespace)
             : m_db(db),
               m_settingsNamespace(settingsNamespace),
               m_iDefaultSortColumn(-1),
               m_eDefaultSortOrder(Qt::AscendingOrder) {
     }
-    virtual ~TrackModel() {}
+
+    virtual ~TrackModel() {
+    }
 
     // These enums are the bits in a bitvector. Any individual column cannot
     // have a value other than 0, 1, 2, 4, or 8!
+    // clang-format off
     enum Capabilities {
-        TRACKMODELCAPS_NONE              = 0x00000,
-        TRACKMODELCAPS_REORDER           = 0x00001,
-        TRACKMODELCAPS_RECEIVEDROPS      = 0x00002,
-        TRACKMODELCAPS_ADDTOPLAYLIST     = 0x00004,
-        TRACKMODELCAPS_ADDTOCRATE        = 0x00008,
-        TRACKMODELCAPS_ADDTOAUTODJ       = 0x00010,
-        TRACKMODELCAPS_LOCKED            = 0x00020,
-        TRACKMODELCAPS_EDITMETADATA      = 0x00040,
-        TRACKMODELCAPS_LOADTODECK        = 0x00080,
-        TRACKMODELCAPS_LOADTOSAMPLER     = 0x00100,
-        TRACKMODELCAPS_LOADTOPREVIEWDECK = 0x00200,
-        TRACKMODELCAPS_REMOVE            = 0x00400,
-        TRACKMODELCAPS_RESETPLAYED       = 0x02000,
-        TRACKMODELCAPS_HIDE              = 0x04000,
-        TRACKMODELCAPS_UNHIDE            = 0x08000,
-        TRACKMODELCAPS_PURGE             = 0x10000,
-        TRACKMODELCAPS_REMOVE_PLAYLIST   = 0x20000,
-        TRACKMODELCAPS_REMOVE_CRATE      = 0x40000,
+        TRACKMODELCAPS_NONE              = 0u,
+        TRACKMODELCAPS_REORDER           = 1u <<  0u,
+        TRACKMODELCAPS_RECEIVEDROPS      = 1u <<  1u,
+        TRACKMODELCAPS_ADDTOTRACKSET     = 1u <<  2u,
+        TRACKMODELCAPS_ADDTOAUTODJ       = 1u <<  3u,
+        TRACKMODELCAPS_LOCKED            = 1u <<  4u,
+        TRACKMODELCAPS_EDITMETADATA      = 1u <<  5u,
+        TRACKMODELCAPS_LOADTODECK        = 1u <<  6u,
+        TRACKMODELCAPS_LOADTOSAMPLER     = 1u <<  7u,
+        TRACKMODELCAPS_LOADTOPREVIEWDECK = 1u <<  8u,
+        TRACKMODELCAPS_REMOVE            = 1u <<  9u,
+        TRACKMODELCAPS_RESETPLAYED       = 1u << 10u,
+        TRACKMODELCAPS_HIDE              = 1u << 11u,
+        TRACKMODELCAPS_UNHIDE            = 1u << 12u,
+        TRACKMODELCAPS_PURGE             = 1u << 13u,
+        TRACKMODELCAPS_REMOVE_PLAYLIST   = 1u << 14u,
+        TRACKMODELCAPS_REMOVE_CRATE      = 1u << 15u,
     };
-    typedef int CapabilitiesFlags; /** Enables us to do ORing */
+    /// Enables the use of bitwise operators
+    typedef unsigned int CapabilitiesFlags;
```

Comment:
Please use QFlags instead:
https://doc.qt.io/qt-5/qflags.html

Maybe it makes sense to rename the enum class from `Capabilities` to `Capability` and the flags from `CapabilitiesFlags` to `Capabilities`.

Reply:
For checking if a flag is set you can then use `caps.testFlag(Capability::Foo)` instead of bitwise operations.

Reply:
Actually, you can still use bitwise operators with QFlags - and that seems to still be the best option if you want to check multiple flags at once.

Reply:
Using QFlags now: https://github.com/mixxxdj/mixxx/pull/2810/commits/bc131a19d85c81af5fe632e7b8add4e249ae45e7

Reply:
> Actually, you can still use bitwise operators with QFlags - and that seems to still be the best option if you want to check multiple flags at once.

I think it will be optimized by the compiler anyway, so let's use the most readable option. 

```cpp
// easy
if (caps.testFlag(Capability::Foo) || caps.testFlag(Capability::Bar)) {}

// requires understanding of bitwise operators
if (caps & (Capability::Foo | Capability::Bar)) {}
```

Reply:
I was talking about this:
```cpp
bool BaseTrackTableModel::hasCapabilities(
        TrackModel::Capabilities caps) const {
    return (getCapabilities() & caps) == caps;
}
```
I don't think there is a similarly concise replacement?